### PR TITLE
feat: activate mls for client via mls feature config [FS-1088]

### PR DESCRIPTION
**Pull request by PatrykBuniX**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-1088" title="FS-1088" target="_blank">TaskFS-1088</a>  [Web] Activate MLS for client via MLS feature config
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

We want to add backward compatibility for the clients that were broken after mistakenly enabling FEATURE_ENABLE_MLS flag on their environment. We want to disable MLS for these clients.

Electron's secretsCrypto global config has changed from `secretsCrypto` to `systemCrypto`. Before passing MLS config config object into the core, we check if a current client is using desktop wrapper and whether the new global is defined and enable MLS only in this case.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by atomrc — CHANGES_REQUESTED** [review]
Code-wise, it's all good. But let's not merge for now as we need a plan to be able to work with the current deployed version of the wrapper

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "@emotion/react": "11.10.4",
     "@wireapp/antiscroll-2": "1.3.1",
     "@wireapp/avs": "8.2.16",
-    "@wireapp/core": "32.1.1",
+    "@wireapp/core": "^33.0.0",
```
Comment (atomrc) [inline package.json]: Should be an exact match, no `^`
```suggestion
    "@wireapp/core": "33.0.0",
```

Comment (PatrykBuniX) [reply package.json]: oops 👍

File: src/script/main/app.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ import {Runtime} from '@wireapp/commons';
 
 import {getLogger, Logger} from 'Util/Logger';
 import {t} from 'Util/LocalizerUtil';
-import {arrayToBase64, checkIndexedDb, createRandomUuid} from 'Util/util';
+import {arrayToBase64, checkIndexedDb, createRandomUuid, isMLSDisabledOnElectron} from 'Util/util';
```
Comment (atomrc) [inline src/script/main/app.ts]: as much as possible, we try not to mention electron in the webapp. It doesn't need to be aware of it. 
Let's call it `supportsMLS`

Comment (PatrykBuniX) [reply src/script/main/app.ts]: sure 👍

File: src/script/util/util.ts [hunk 3]
```diff
@@ -398,3 +398,6 @@ export const getSelectionPosition = (element: HTMLTextAreaElement, currentMentio
 
   return {newEnd, newStart};
 };
+
+// temporary hack that disables mls for old 'broken' desktop clients, see https://github.com/wireapp/wire-desktop/pull/6094
+export const isMLSDisabledOnElectron = Runtime.isDesktopApp() && !window.systemCrypto;
```
Comment (atomrc) [inline src/script/util/util.ts]: I think this condition will disable mls for webapp not in electron

```suggestion
export const supportsMLS = !Runtime.isDesktopApp() || window.systemCrypto;
```

**Review by atomrc — APPROVED** [review]

**Review by atomrc — APPROVED** [review]

**Pull request merged**